sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala

```diff
@@ -31,7 +31,7 @@ import org.apache.spark.sql.internal.SQLConf
  * input partition ordering requirements are met.
  */
 case class EnsureRequirements(conf: SQLConf) extends Rule[SparkPlan] {
-  private def defaultNumPreShufflePartitions: Int = conf.numShufflePartitions
+  private lazy val defaultNumPreShufflePartitions: Int = conf.numShufflePartitions
```

Comment:
Actually it is to be sure that value does not change dynamically across different calls.